File: java/23_Merge_k_Sorted_Lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeKLists(ListNode[] lists) {
        ArrayList<Integer> nodes = new ArrayList<Integer>();
        ListNode head = new ListNode(0);
        ListNode point = head;
        for (ListNode list : lists) {
            while (list != null) {
                nodes.add(list.val);
                list = list.next;
            }
        }

        Collections.sort(nodes);   
        for (int node: nodes) {
            point.next = new ListNode(node);
            point = point.next;
        }
        
        return head.next;
    }
}

File: php/0000/23_Merge_k_Sorted_Lists.php
<?php

/**
 * Definition for a singly-linked list.
 * class ListNode {
 *     public $val = 0;
 *     public $next = null;
 *     function __construct($val = 0, $next = null) {
 *         $this->val = $val;
 *         $this->next = $next;
 *     }
 * }
 */
class Solution
{

    /**
     * @param ListNode[] $lists
     * @return ListNode
     */
    function mergeKLists($lists)
    {
        $nodes = [];
        $head = $point = new ListNode(0);
        foreach ($lists as $list) {
            while ($list != null) {
                $nodes[] = $list->val;
                $list = $list->next;
            }
        }

        sort($nodes);
        foreach ($nodes as $node) {
            $point->next = new ListNode($node);
            $point = $point->next;
        }

        return $head->next;
    }

}
